pkg/storage/memory-storage/memory-storage.go

```diff
@@ -0,0 +1,113 @@
+package memory_storage
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/DanilLagunov/jokes-api/pkg/models"
+)
+
+var JokeNotFound = errors.New("joke not found")
+
+type MemoryStorage []models.Joke
+
+func NewMemoryStorage() MemoryStorage {
+	var storage MemoryStorage
+	parseJSON("../pkg/storage/memory-storage/reddit_jokes.json", &storage)
+	return storage
+}
+
+func (s MemoryStorage) GetJokes() ([]models.Joke, error) {
+	return s, nil
+}
+
+func (s MemoryStorage) AddJoke(title, body string) error {
+	var id string
+CHECK:
+	id = models.GenerateID()
+	for i := 0; i < len(s); i++ {
+		if id == s[i].ID {
+			goto CHECK
+		}
+	}
+
+	joke := models.NewJoke(id, title, body, 0)
+	s = append(s, joke)
+
+	rawDataOut, err := json.MarshalIndent(&s, "", "   ")
+	if err != nil {
+		log.Fatal("JSON marshalling failed: ", err)
+	}
+	err = ioutil.WriteFile("../pkg/storage/memory-storage/reddit_jokes.json", rawDataOut, 0)
```

Comment:
you can create variable in MemoryStorage that should keep path to the file and not to write it here.